### PR TITLE
[Doc] Enable Mermaid diagrams in our documentation

### PR DESCRIPTION
**Pull request by Oscilloscope98**
### 1. Why the change?
https://github.com/intel-analytics/BigDL/issues/6439

### 2. Summary of the change 

- Enable Mermaid diagram in our documentation
- Resolve conflicts between nbsphinx and sphinxcontrib-mermaid caused by require.js, keeping the full functionalities of nbsphinx including [Interactive Widgets](https://nbsphinx.readthedocs.io/en/0.8.7/code-cells.html#Interactive-Widgets-(HTML-only)) (refer to [here](https://github.com/spatialaudio/nbsphinx/issues/678) and [here](https://github.com/spatialaudio/nbsphinx/issues/519#issuecomment-731828712))
  - Disable mermaid.js loaded from sphinxcontrib-mermaid
  - Load mermaid.js before require.js
- Add how to use Mermaid diagram in documentation guide
  image
  image

### 4. How to test?
- [x] Document test: https://yuwentestdocs.readthedocs.io/en/enable-mermaid-js/doc/UserGuide/documentation.html

### 5. New dependencies

- New Sphinx and js depencies
  - [sphinxcontrib-mermaid](https://github.com/mgaitan/sphinxcontrib-mermaid) ([license](https://github.com/mgaitan/sphinxcontrib-mermaid/blob/master/LICENSE.rst))
  - https://cdnjs.cloudflare.com/ajax/libs/mermaid/9.2.0/mermaid.min.js

### CODE REVIEW

**Comment by Oscilloscope98** [comment]
As the loaded [mermaid.min.js](https://unpkg.com/browse/mermaid@9.2.0/dist/mermaid.min.js) is about 1.09MB, when the network connection is slow, the situation that Mermaid code first show as plain texts, and then become diagrams may happens.

In further PR, we may want to load it from local folder to improve the behavior.

File: docs/readthedocs/source/_templates/layout.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{% extends "!layout.html" %}
+
+{% block scripts %}
+<script src="https://cdnjs.cloudflare.com/ajax/libs/mermaid/9.2.0/mermaid.min.js" integrity="sha512-bAUSYGDF/K+6ME3orRcyGh86wxE7kBRunb1tPd4btqx+iZrS3CWaxil1KNUNsVw2OlqQJvXw3id+jsPflaK49g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
+{{ super() }}
+{% endblock %}
```
Comment (qiyuangong) [inline docs/readthedocs/source/_templates/layout.html:6]: Add an empty line at the end

File: docs/readthedocs/requirements-doc.txt [hunk 2]
```diff
@@ -40,4 +40,5 @@ nbsphinx==0.8.9
 ipython==7.34.0
 sphinx-design==0.2.0
 nbsphinx-link==1.3.0
-pydata-sphinx-theme==0.11.0
\ No newline at end of file
+pydata-sphinx-theme==0.11.0
```
Comment (qiyuangong) [inline docs/readthedocs/requirements-doc.txt:43]: [pydata-sphinx-theme](https://github.com/pydata/pydata-sphinx-theme) is BSD

**Review by qiyuangong — APPROVED** [review]
LGTM

File: docs/readthedocs/source/_templates/layout.html (hunk 1 above)
Comment (Oscilloscope98) [reply docs/readthedocs/source/_templates/layout.html]: Have added.

**Pull request merged**